crates/autopilot/src/run_loop.rs

```diff
@@ -94,18 +97,27 @@ impl RunLoop {
         solutions.sort_unstable_by_key(|solution| solution.1.score);
         let competition_simulation_block = self.current_block.borrow().number;
 
-        let events: Vec<_> = solutions
-            .iter()
-            .flat_map(|(_, s)| s.orders.iter().map(|o| (*o, OrderEventLabel::Considered)))
-            .collect();
-
         // TODO: Keep going with other solutions until some deadline.
         if let Some((index, solution)) = solutions.last() {
             // The winner has score 0 so all solutions are empty.
             if solution.score == 0.into() {
                 return;
             }
 
+            tracing::info!(url = %self.drivers[*index].url, "revealing with solver");
+            let revealed = match self.reveal(id, &self.drivers[*index]).await {
+                Ok(result) => result,
+                Err(err) => {
+                    tracing::error!(?err, "solver {index} failed to reveal");
+                    return;
+                }
+            };
+
+            let events = revealed
+                .orders
+                .iter()
+                .map(|o| (*o, OrderEventLabel::Considered))
+                .collect::<Vec<_>>();
```

Comment:
Do you know why we need this information? At least I don't see a purpose to know if some order was included in some solution that did not win?
Is it worth forcing the `/solve` to return this data just for debugging purposes? 
Reveal should rarely fail since it is simple -> doesn't even communicate to underlying solver, it just reads already saved settlement and returns simple data -> should be quite fast.

Reply:
> Do you know why we need this information?

It's part of our SLI initiative which aims to give us a detailed overview where in our systems orders spend a lot of time. If an order is not part of the winning solution but some runner up we could estimate with this information how much we could gain from letting multiple solver submit disjoint settlements for example.

> Is it worth forcing the /solve to return this data just for debugging purposes?

Hmm, you have a point. This was not part of the initial API design and we probably shouldn't change that because we want to have some debugging information.
On the other hand this doesn't seem like a big ask from a solver's perspective. They already have to estimate a score for their solution so surely they must have an idea which orders this solution would settle.
I think it's fine to not have that for now.